compatibility_tests/test_file_compatibility.py

```diff
@@ -0,0 +1,184 @@
+
+
+def fetch_package_versions(package_name):
+    """
+    Request a package's available versions from pypi.org metadata.
+    """
+    content = urllib.request.urlopen("https://pypi.org/pypi/{}/json".format(package_name)).read()
+    version_strings = json.loads(content)["releases"].keys()
+    return [StrictVersion(v) for v in version_strings if v not in BAD_VERSIONS and v >= MIN_VERSION]
+
+
+def fetch_latest_patch_versions(package_name):
+    """
+    Return the latest patch version within each of the package's
+    minor versions.
+    """
+    key_fn = lambda v: v.version[0:2]
+
+    versions = sorted(fetch_package_versions(package_name), key=key_fn)
+    return [max(group) for _, group in groupby(versions, key=key_fn)]
+
+
+# Enable internet here, otherwise pytest_remotedata will complain
+# (and @pytest.mark.remote_data doesn't work on non-test methods).
+with internet_temporarily_enabled():
+    PATCH_VERSIONS = fetch_latest_patch_versions("asdf")
+
+
+def env_run(env_path, command, *args, **kwargs):
+    """
+    Run a command on the context of the virtual environment at
+    the specified path.
+    """
+    return subprocess.run([env_path/"bin"/command] + list(args), **kwargs).returncode == 0
+
+
+def env_check_output(env_path, command, *args):
+    """
+    Run a command on the context of the virtual environment at
+    the specified path, and return the output.
+    """
+    return subprocess.check_output([env_path/"bin"/command] + list(args)).decode("utf-8").strip()
+
+
+def get_supported_versions(env_path):
+    """
+    Get ASDF Standard versions that are supported by the asdf library
+    installed in the specified virtual environment.
+    """
+    script = r"""import asdf; print("\n".join(str(v) for v in asdf.versioning.supported_versions))"""
+    output = env_check_output(env_path, "python", "-c", script)
+    return [asdf.versioning.AsdfVersion(v) for v in output.split("\n")]
+
+
+def get_installed_version(env_path):
+    """
+    Get the version of the asdf library installed in the specified
+    virtual environment.
+    """
+    script = r"""import asdf; print(asdf.__version__)"""
+    return StrictVersion(env_check_output(env_path, "python", "-c", script))
+
+
+@pytest.fixture(scope="module", params=PATCH_VERSIONS)
+def asdf_version(request):
+    """
+    The (old) version of the asdf library under test.
+    """
+    return request.param
+
+
+@pytest.fixture(scope="module")
+def env_path(asdf_version, tmp_path_factory):
+    """
+    Path to the virtualenv where the (old) asdf library is installed.
+    """
+    path = tmp_path_factory.mktemp("asdf-{}-env".format(asdf_version), numbered=False)
+
+    virtualenv.cli_run([str(path)])
+
+    assert env_run(path, "pip", "install", "asdf=={}".format(asdf_version), capture_output=True), (
+        "Failed to install asdf version {}".format(asdf_version)
+    )
+
+    return path
+
+
+@pytest.fixture(autouse=True)
+def change_tmpdir(tmpdir):
```

Comment:
Perhaps this could function could be called `pushd`.?

Reply:
Haha, I can hardly [disagree](https://github.com/spacetelescope/jwst/pull/4567#discussion_r378637258)